theories/algebra/IntDiv.ec

```diff
@@ -864,9 +864,37 @@ move=> lt_pn; rewrite {1}(divz_eq i (k^n)); rewrite -addrA; congr.
 by rewrite {1}(divz_eq (i %% k^n) (k^p)) modz_dvd_pow.
 qed.
 
-lemma nosmt modz_pow2_div n p i: 0 <= i => 0 <= p <= n =>
-  (i %% 2^n) %/ 2^p = (i %/ 2^p) %% 2^(n-p).
-proof. admitted.
+lemma nosmt divzMr i a b :  
+  0 <= a => 0 <= b => i %/ (a * b) = i %/ a %/ b.
+proof.
+move=> ge0_a ge0_b.
+case (a = 0); first by smt(div0z). 
```

Comment:
It's possibly slightly better for readability to introduce `gt0_a` here, rather than after the second case.

Reply:
I assume you mean `neq0_a` instead of `gt0_a`?

Reply:
Yes, sorry.